compiler/expressions.go

```diff
@@ -592,7 +592,7 @@ func (c *funcContext) translateExpr(expr ast.Expr) *expression {
 				declaredFuncRecv := sel.Obj().(*types.Func).Type().(*types.Signature).Recv().Type()
 				if typesutil.IsJsObject(declaredFuncRecv) {
 					globalRef := func(id string) string {
-						if recv.String() == "$global" && id[0] == '$' {
+						if recv.String() == "$global" && id[0] == '$' && len(id) > 1 {
```

Comment:
I tried to read the code outside and found it hard to see what guarantees that length of `id` is always at least one (so that the `id[0]` access doesn't panic).